rasa/nlu/components.py

```diff
@@ -87,16 +87,42 @@ def validate_required_components_from_data(
     pipeline: List["Component"], data: TrainingData
 ):
 
-    response_selector_exists = False
-    for component in pipeline:
-        # check if a response selector is part of NLU pipeline
-        if MESSAGE_RESPONSE_ATTRIBUTE in component.provides:
-            response_selector_exists = True
+    if data.regex_features and not any(
+        [component.name == "RegexFeaturizer" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data with regexes, but "
+            "your NLU pipeline does not include a RegexFeaturizer. "
+            "To featurize regexes, add a RegexFeaturizer to your pipeline."
+        )
+
+    if data.lookup_tables and not any(
+        [component.name == "CRFEntityExtractor" for component in pipeline]
```

Comment:
Is this the correct one? AFAIK you can use it without entities at all -- lookup tables are just big regexes, i.e. they would need the regexfeaturizer.  can you ask someone with more NLU experience?

Reply:
yea I wasn't sure either, any recommendations on who to ask about this?

Reply:
vova or akela probably, just ask in the channel it'll be faster

Reply:
@erohmensing do you think it makes sense to write the new code like this or add a conditional on top of this to know which part is missing?

```
if data.regex_features and not any(
        [
            component.name == "RegexFeaturizer"
            and component.name == "CRFEntityExtractor"
            for component in pipeline
        ]
    ):
        warnings.warn(
            "You have defined training data with regexes, but "
            "your NLU pipeline does not include a RegexFeaturizer or CRFEntityExtractor. "
            "To featurize regexes, add a RegexFeaturizer or CRFEntityExtractor to your pipeline."
        )
```

Reply:
well this code won't work -- 
```
        [
            component.name == "RegexFeaturizer"
            and component.name == "CRFEntityExtractor"
            for component in pipeline
        ]
```
there is no component that is named both regexfeaturizer and crfentityfeaturizer, so this will generate false for every component, therefore this warning will always show.

Your decision if you think there should be extra logic or not, but in any case you should make it specific that "If you want to featurize regexes *for entity extraction*" you need a CRFEntityExtractor.

Reply:
Sidenote, is there already a warning in place if you define entities in markdown format but don't have a CRFEntityExtractor?

Reply:
@erohmensing any opinion on how we might go about this?  I'm torn which way to do it, I like the idea of less conditionals and just stating to check both things but not sure.

Reply:
> Sidenote, is there already a warning in place if you define entities in markdown format but don't have a CRFEntityExtractor?

Not sure I'll have to check on that

Reply:
> well this code won't work --
> 
> ```
>         [
>             component.name == "RegexFeaturizer"
>             and component.name == "CRFEntityExtractor"
>             for component in pipeline
>         ]
> ```
> 
> there is no component that is named both regexfeaturizer and crfentityfeaturizer, so this will generate false for every component, therefore this warning will always show.
> 
> Your decision if you think there should be extra logic or not, but in any case you should make it specific that "If you want to featurize regexes _for entity extraction_" you need a CRFEntityExtractor.

I'll have to double check but this is a list of pipeline stuff so if they are both in there then this should be fine which is what I was testing initially here with this code.

Reply:
@erohmensing I have updated the code to test and it appears to be working if I have the following for example in my `config.yml`:

```
# Configuration for Rasa NLU.
# https://rasa.com/docs/rasa/nlu/components/
language: en
pipeline:
- name: "WhitespaceTokenizer"
  intent_split_symbol: "_"
- name: "CountVectorsFeaturizer"
- name: "EmbeddingIntentClassifier"
- name: "ResponseSelector"
- name: "CRFEntityExtractor"
# - name: "RegexFeaturizer"
- name: "EntitySynonymMapper"
```

Then I get the message back:

```
UserWarning: You have defined training data with regexes, but your NLU pipeline does not include a RegexFeaturizer or CRFEntityExtractor. To featurize regexes, add a RegexFeaturizer or CRFEntityExtractor to your pipeline.
```

If I comment out both the Regex and CRF I get the message as well, so I think this is working right, my only question is do we need to have a specific check for RegexFeaturizer by itself, I don't think so but just verifying.

Reply:
@erohmensing ok how does this latest update look to you?  Here is the code to save some time looking in the commit:

```
# Check for Regex data but RegexFeaturizer not enabled
    if data.regex_features and not any(
        [component.name == "RegexFeaturizer" for component in pipeline]
    ):
        warnings.warn(
            "You have defined training data with regexes, but "
            "your NLU pipeline does not include an RegexFeaturizer. "
            "To featurize regexes for entity extraction, you need to have RegexFeaturizer in your pipeline."
        )
```